pkgs/applications/science/logic/yices/default.nix

```diff
@@ -25,12 +26,16 @@ stdenv.mkDerivation rec {
   # Usual shenanigans
   patchPhase = "patchShebangs tests/regress/check.sh";
 
-  # Includes a fix for the embedded soname being libyices.so.2.5, but
-  # only installing the libyices.so.2.5.x file.
+  # Includes a fix for the embedded soname being libyices.so.X.Y, but
+  # only installing the libyices.so.X.Y.Z file.
   installPhase = let
     ver_XdotY = lib.versions.majorMinor version;
   in ''
       make install LDCONFIG=true
+      # guard against packaging of unstable versions: they
+      # have a soname of hext (not current) release.
+      echo "Checking expected library version to be ${version}"
+      [ -f $out/lib/libyices.so.${version} ]
```

Comment:
This is incorrect on darwin and breaks yices on that platform: <https://hydra.nixos.org/build/156534343/nixlog/1>

Reply:
I'll have a look and send a fix.

Reply:
```diff
diff --git a/pkgs/applications/science/logic/yices/default.nix b/pkgs/applications/science/logic/yices/default.nix
index c5fc3a73837..992d40d594a 100644
--- a/pkgs/applications/science/logic/yices/default.nix
+++ b/pkgs/applications/science/logic/yices/default.nix
@@ -30,13 +30,14 @@ stdenv.mkDerivation rec {
   # only installing the libyices.so.X.Y.Z file.
   installPhase = let
     ver_XdotY = lib.versions.majorMinor version;
+    inherit (stdenv.hostPlatform) extensions;
   in ''
       make install LDCONFIG=true
       # guard against packaging of unstable versions: they
       # have a soname of hext (not current) release.
       echo "Checking expected library version to be ${version}"
-      [ -f $out/lib/libyices.so.${version} ]
-      ln -sfr $out/lib/libyices.so.{${version},${ver_XdotY}}
+      [ -f $out/lib/libyices.${extensions.sharedLibrary}.${version} ]
+      ln -sfr $out/lib/libyices.${extensions.sharedLibrary}.{${version},${ver_XdotY}}
   '';
 
   meta = with lib; {

```

started this, but untested. Seems like this is incorrect because *I think* versioned libs are `libyices.${version}.dylib`…

Reply:
Won't have time to look into it more the next two days

Reply:
Proposed fix as https://github.com/NixOS/nixpkgs/pull/143830